addons/zebra-striping/userscript.js

```diff
@@ -1,69 +1,82 @@
+    if (
+      // addon is enabled
+      !addon.self.disabled &&
+      // not in the flyout
+      !this.isInFlyout &&
+      // not a shadow
+      !this.isShadow() &&
+      // not an insertion marker
+      !this.isInsertionMarker() &&
+      // does not have parent
+      this.getParent() === null
+    ) {
+      // getDescendants() returns blocks in an order such that each block's parent
+      // will always come before that block (except the first block which has no parent).
       for (const block of this.getDescendants()) {
         const parent = block.getSurroundParent();
 
-        let isStriped = false;
-        if (parent) {
-          if (block.isShadow()) {
-            isStriped = !!stripeState.get(parent);
-          } else if (parent.getColour() === block.getColour()) {
-            isStriped = !stripeState.get(parent);
-          }
+        const shouldStripe =
+          // not a insertion marker
+          !block.isInsertionMarker() &&
+          // has a parent
+          parent &&
+          // shadow blocks should be striped if it's parent is striped. otherwise, it should be opposite of parent.
+          (block.isShadow() ? parent.saStriped : !parent.saStriped) &&
+          // parent and child are same color (even shadows match block color)
+          // we dont check category because other addons can make blocks the same color.
+          // if the block was stripped we need to look at it's original color in order to get it's real color.
+          parent.getColour() === (block.saOrginalColour ? block.saOrginalColour[0] : block.getColour());
+
+        // if the block's stripe state is correct, no need to update its state.
+        if (shouldStripe === block.saStriped) continue;
+
+        if (!shouldStripe && block.saOrginalColour) {
+          block.setColour(...block.saOrginalColour);
+          block.saOrginalColour = null;
+        } else if (shouldStripe && !block.saOrginalColour) {
+          const color = block.getColour();
+          const secondary = block.getColourSecondary();
+          const tertiary = block.getColourTertiary();
+          const quaternary = block.getColourQuaternary();
+          block.saOrginalColour = [color, secondary, tertiary, quaternary];
+
+          const stripedColors = block.saOrginalColour.map((c) => "#" + tinycolor(c).lighten(amount).toHex());
+          block.setColour(...stripedColors);
         }
-        stripeState.set(block, isStriped);
 
-        const elements = [block.svgPath_];
+        // field strokes don't automatically update when setColour is called.
+        // so manually update field dropdowns to match parent color
         for (const input of block.inputList) {
-          if (input.outlinePath) {
-            elements.push(input.outlinePath);
-          }
           for (const field of input.fieldRow) {
-            if (field.fieldGroup_) {
-              elements.push(field.fieldGroup_);
+            if (field instanceof ScratchBlocks.FieldDropdown) {
```

Comment:
```suggestion
            if (field instanceof ScratchBlocks.FieldDropdown && field.box_) {
```

Reply:
why check .box_ (so i can put it in the comments)

Reply:
There's no box if the dropdown is in a shadow: https://github.com/scratchfoundation/scratch-blocks/blob/develop/core%2Ffield_dropdown.js#L139